src/main/java/com/minecolonies/core/entity/ai/workers/production/EntityAIQuarrier.java

```diff
@@ -359,6 +359,20 @@ protected boolean skipBuilding(final BlueprintPositionInfo info, final BlockPos
                  || DONT_TOUCH_PREDICATE.test(info, pos, handler);
     }
 
+    @Override
+    protected boolean skipClearing(final BlueprintPositionInfo info, final BlockPos pos, final IStructureHandler handler)
+    {
+        if (super.skipClearing(info, pos, handler))
+        {
+            return true;
+        }
+        final BlockState state = handler.getWorld().getBlockState(pos);
+        // The blocks have been placed earlier by the quarrier. E.g. fences and walls can have the wrong blockstate
+        // (due to connecting to yet to be mined blocks), which means the block state is incorrect, and they would be mined by the quarrier,
+        // only to be replaced after the next restart. Instead, skip them now if the block itself is correct
+        return state.getBlock() == info.getBlockInfo().getState().getBlock();
```

Comment:
Only thing I'm unsure about, block classes don't seem to have an equals method, can you guarantee these are identical instances?

Reply:
That is how blocks work afaik. A different instance means a different block altogether

Reply:
That is, each dirt block in the world would reference to a block state of the same block instance (in this case Blocks.DIRT). Dirt and cobblestone would both be instances of the base Block class, etc.

Reply:
Anything with registry can/should be considered for instance equality

Reply:
ye, this is fine